tests/unit/s2n_cipher_suite_match_test.c

```diff
@@ -850,5 +850,523 @@ int main(int argc, char **argv)
+                    /* Second chacha20 ciphersuite and is negotiated */
+                    &s2n_ecdhe_rsa_with_chacha20_poly1305_sha256,   
+                };
+
+                cipher_preferences = (struct s2n_cipher_preferences) {
+                    .count = s2n_array_len(test_cipher_suite_list),
+                    .suites = test_cipher_suite_list,
+                    .allow_chacha20_boosting = true
+                };
+
+                /*
+                 * Client signalled chacha20 boosting.
+                 * if chacha20_boosting is enabled by the server:
+                 *      TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305_SHA256 > TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA384 
+                 * else:
+                 *      TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA384 > TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305_SHA256 
+                 */
+                uint8_t ecdhe_chacha20_boosted_wire[] = {
+                    TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305_SHA256,
+                    TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA384,
+                };
+                uint8_t count = sizeof(ecdhe_chacha20_boosted_wire) / S2N_TLS_CIPHER_SUITE_LEN;
+
+                /* Verify ecdhe ecdsa chacha20 ciphersuite chosen with chacha20 boosting enabled */
+                EXPECT_SUCCESS(s2n_set_cipher_as_tls_server(connection, ecdhe_chacha20_boosted_wire, count));
+                EXPECT_EQUAL(connection->secure->cipher_suite, &s2n_ecdhe_ecdsa_with_chacha20_poly1305_sha256);
+
+                /* Sanity check: non-chacha20 aes_256_cbc ciphersuite chosen without chacha20 boosting enabled */
+                cipher_preferences.allow_chacha20_boosting = false;
+                EXPECT_SUCCESS(s2n_set_cipher_as_tls_server(connection, ecdhe_chacha20_boosted_wire, count));
+                EXPECT_EQUAL(connection->secure->cipher_suite, &s2n_ecdhe_rsa_with_aes_256_cbc_sha384);
+
+            }
+
+            /* Server does not negotiate the client's most preferred chacha20 ciphersuite */
+            {
+                DEFER_CLEANUP(struct s2n_connection *connection = s2n_connection_new(S2N_SERVER), s2n_connection_ptr_free);
+                EXPECT_NOT_NULL(connection);
+                connection->security_policy_override = &security_policy;
+                connection->kex_params.server_ecc_evp_params.negotiated_curve = s2n_all_supported_curves_list[0];
+                EXPECT_SUCCESS(s2n_connection_set_all_protocol_versions(connection, S2N_TLS12));
+                EXPECT_SUCCESS(s2n_connection_set_config(connection, server_config));
+
+                static struct s2n_cipher_suite *test_cipher_suite_list[] = {
+                    /* Skipped because not a chacha20 ciphersuite */
+                    &s2n_ecdhe_ecdsa_with_aes_128_cbc_sha256,       
+                    /* Skipped because not a chacha20 ciphersuite. If chacha20 boosting is off then this is negotiated.*/
+                    &s2n_ecdhe_rsa_with_aes_256_cbc_sha384,         
```

Comment:
If chacha20 boosting is turned off, why isn't s2n_ecdhe_ecdsa_with_aes_128_cbc_sha256 chosen? Is this important to the test?

Reply:
Because it's not in the client wire. This just shows that server will follow server-preferences when chacha20 boosting is off.

Reply:
Your comment should indicate that. But what's the point of s2n_ecdhe_ecdsa_with_aes_128_cbc_sha256 if it's not even an option? Why is it important to this test?

Reply:
Unless this is a different test, I still see it in the latest revision: https://github.com/aws/s2n-tls/blob/875ebd02300511e0204ccb30554c027a60b26e72/tests/unit/s2n_cipher_suite_match_test.c#L944-L947